platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/status/PlatformStatusComponent.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.system.status;
+
+import com.swirlds.base.state.Startable;
+import com.swirlds.base.state.Stoppable;
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.notification.NotificationEngine;
+import com.swirlds.common.system.status.actions.PlatformStatusAction;
+import com.swirlds.common.system.status.actions.TimeElapsedAction;
+import com.swirlds.common.threading.framework.QueueThread;
+import com.swirlds.common.threading.framework.config.QueueThreadConfiguration;
+import com.swirlds.common.threading.framework.config.QueueThreadMetricsConfiguration;
+import com.swirlds.common.threading.manager.ThreadManager;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+
+/**
+ * An asynchronous component which manages platform status.
+ * <p>
+ * This object wraps a {@link PlatformStatusStateMachine}, which contains the actual state machine logic.
+ */
+public class PlatformStatusComponent implements PlatformStatusGetter, Startable, Stoppable {
+    /**
+     * A source of time
+     */
+    private final Time time;
+
+    /**
+     * The thread to handle incoming {@link PlatformStatusAction}s
+     */
+    private final QueueThread<PlatformStatusAction> handleThread;
```

Comment:
In other classes, queue threads are usually referred to as a `queue` or a `queueThread`, not a `thread`. The interface it exposes is that of a queue and not a thread.

Reply:
Fixed